Ban org.ow2.asm:asm in non-deployment modules

Before https://github.com/quarkiverse/quarkus-cxf/pull/487, we were banning `org.ow2.asm:asm` in some modules. We should investigate whether we should/can ban it after https://github.com/quarkiverse/quarkus-cxf/pull/487 and why.

When banned, it would have to be excluded from the follwing BOM entries:

```
[ERROR]     io.quarkiverse.cxf:quarkus-cxf:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-features-logging:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-features-metrics:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-transports-http-hc5:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     org.apache.cxf:cxf-rt-ws-security:3.5.3:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     org.apache.cxf:cxf-rt-ws-mex:3.5.3:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-ws-security:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
```

As a first step, we can do the above exclusions and check whether some tests fail.

--- a/extensions/pom.xml
+++ b/extensions/pom.xml
@@ -21,52 +21,40 @@
         <module>transports-http-hc5</module>
     </modules>
 
-    <build>
-        <plugins>
-            <plugin>
-                <groupId>org.codehaus.gmaven</groupId>
-                <artifactId>groovy-maven-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>set-enforce-non-deployment-property</id>
-                        <inherited>false</inherited>
-                        <goals>
-                            <goal>execute</goal>
-                        </goals>
-                        <phase>process-resources</phase>
-                        <configuration>
-                            <source>
-                                if (project.artifactId.endsWith('-deployment')) {
-                                    properties['enforce-non-deployment.skip'] = true;
-                                }
-                            </source>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-enforcer-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>enforce-non-deployment</id>
-                        <configuration>
-                            <skip>${enforce-non-deployment.skip}</skip>
-                            <rules>
-                                <bannedDependencies>
-                                    <excludes>
-                                        <!-- <exclude>org.ow2.asm:asm</exclude> -->
-                                    </excludes>
-                                </bannedDependencies>
-                            </rules>
-                        </configuration>
-                        <goals>
-                            <goal>enforce</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
-        </plugins>
-    </build>
-
+    <profiles>
+        <profile>
+            <id>non-deployment-bans</id>
+            <activation>
+                <file>
+                    <exists>${basedir}/src/main/resources/META-INF/quarkus-extension.yaml</exists>
+                </file>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-enforcer-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>enforce-non-deployment</id>
+                                <configuration>
+                                    <skip>${enforce-non-deployment.skip}</skip>
+                                    <rules>
+                                        <bannedDependencies>
+                                            <excludes>
+                                                <exclude>org.ow2.asm:asm</exclude>
+                                            </excludes>
+                                        </bannedDependencies>
+                                    </rules>
+                                </configuration>
+                                <goals>
+                                    <goal>enforce</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/pom.xml
+++ b/pom.xml
@@ -54,7 +54,6 @@
         <supported-maven.versions>[3.6.2,)</supported-maven.versions>
 
         <cq-maven-plugin.version>3.1.1</cq-maven-plugin.version>
-        <groovy-maven-plugin.version>2.1.1</groovy-maven-plugin.version>
         <maven-enforcer-plugin.version>3.0.0-M3</maven-enforcer-plugin.version>
         <maven-eclipse-plugin.version>2.10</maven-eclipse-plugin.version>
         <maven-idea-plugin.version>2.2.1</maven-idea-plugin.version>
@@ -137,11 +136,6 @@
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-failsafe-plugin</artifactId>
                     <version>${maven-failsafe-plugin.version}</version>
-                </plugin>
-                <plugin>
-                    <groupId>org.codehaus.gmaven</groupId>
-                    <artifactId>groovy-maven-plugin</artifactId>
-                    <version>${groovy-maven-plugin.version}</version>
                 </plugin>
                 <plugin>
                     <groupId>io.quarkus</groupId>

